slurp.py

```diff
@@ -1002,35 +1024,30 @@ def matches( rule, kwargs={} ):
         if pfn_lists.get(runseg,None)==None:
             pfn_lists[runseg]=[]
 
-        # Build list of PFNs via direct lookup and append the results
-        try:
-            if rule.direct:
+        # Build list of PFNs via direct lookup and append the results... TODO... both branches of this IF statement look identical...
 
-                pfn_lists[runseg] = [lfn2pfn[lfn] for lfn in lfns] 
+        if rule.direct:
 
-                # Build list of PFNs via filecatalog lookup if direct path has not been specified
-            if rule.direct==None:            
+            pfn_lists[runseg] = [lfn2pfn[lfn] for lfn in lfns] 
 
-                pfn_lists[runseg] = [lfn2pfn[lfn] for lfn in lfns]
+        # Build list of PFNs via filecatalog lookup if direct path has not been specified
+        if rule.direct==None:            
 
-        except KeyError:
-            print( "No PFN for all LFNs in the input query.")
-            print(f"direct_path: {str(rule.direct)}")
-            print( "    ... if it is None, you should specify the directory paths where input files can be found")
-            print( "        in the input query.")
-            print( "    ... if it specifies one or more directories, then your list is incomplete, or there are missing input files.")
-            raise KeyError
+            pfn_lists[runseg] = [lfn2pfn[lfn] for lfn in lfns]
 
+            
     INFO(f"... {len(pfn_lists.keys())} pfn lists")
 
     # 
     # The production setup will be unique based on (1) the specified analysis build, (2) the specified DB tag,
     # and (3) the hash of the local github repository where the payload scripts/macros are found.
     #
-    repo_dir  = payload #'/'.join(payload.split('/')[1:]) 
+    repo_dir  = payload 
```

Comment:
may need to forcibly strip off whitespace and/or a trailing '/' from the path.  TBD.